### DISCUSSION TITLE
Example of Malformed Data Indicated by ekg-clean-db

### DISCUSSION THREAD

**Discussion by jayrajput** [ahyatt/ekg#36] [Q&A] [answered]
The current documentation of `ekg-clean-db` hints on malformed data. What will be an example of such a data?

```
ekg-clean-db is an interactive Lisp closure in ‘ekg.el’.

(ekg-clean-db)

Clean all useless or malformed data from the database.
Some of this are tags which have no uses, which we consider
useless. This will always make a backup, regardless of backup
settings, and will not delete any backups, regardless of other
settings.
```

Further will be better to create a separate function called `ekg-delete-unused-tags`. It feels odd/scary calling `ekg-clean-db` to remove unused tags. I recently moved from org-roam to ekg and keep on reaching out to calling `ekg-clean-db`.

**Comment by ahyatt** [comment, marked as answer]
Good call out, thanks. I've left it somewhat vague because I'll probably add more things over time, but I'll add more text so you know exactly what it is doing and why.   But I wouldn't call `ekg-clean-db` that often unless you are noticing tags in completions that have no content, or things like that, and you think they shouldn't exist.  But, maybe it's OK for it to exist, maybe you want it use it later.  In reality, it shouldn't matter much, so I'd recommend ignoring this function unless you find your database size an issue.
